decide/src/main/java/group17/LicAnalyzer.java

```diff
@@ -203,7 +203,45 @@ public boolean lic7(InputHandler input) {
         return false;
     }
 
+    /**
+     * This function calculates Launch Interceptor Condition (LIC) number 8
+     * Utilizes helper methods in GeometryUtils.java to calculate the smallest possible radius between 3 points.
+     * The first two points are separated by A_PTS.
+     * The last two points are separated by B_PTS.
+     *
+     * @param input.NUMPOINTS,RADIUS1,X_COORD,Y_COORD,A_PTS,B_PTS
+     * @return true if three points, separated by A_PTS and B_PTS cannot be contained on or within a circle of radius RADIUS1
+     */
     public boolean lic8(InputHandler input) {
+
+        double[] X = input.X_COORD;
+        double[] Y = input.Y_COORD;
+        int A = input.A_PTS;
+        int B = input.B_PTS;
+        
+        if (input.NUMPOINTS < 2 || input.NUMPOINTS > 100) {
+            throw new IllegalArgumentException("NUMPOINTS must be between 2 and 100.");
```

Comment:
I see that your verifying `NUMPOINTS` to be in range [2, 100], does this together check together with checks below handle the requirement that `LIC8` should throw if `A_PTS + B_PTS > NUMPOINTS + 3`?

Reply:
Yes, the ```A_PTS + B_PTS > NUMPOINTS + 3``` becomes implicit.